src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/SymbolTable.cs

```diff
@@ -898,11 +898,7 @@ private static List<object> BuildDeclarationChain(Type callingType)
                     ParameterInfo[] parameters = methodBase.GetParameters();
 
                     bool bAdded = false;
-#if UNSUPPORTEDAPI
-                    foreach (MethodInfo methinfo in Enumerable.Where(t.DeclaringType.GetRuntimeMethods(), m => m.MetadataToken == methodBase.MetadataToken))
-#else
                     foreach (MethodInfo methinfo in Enumerable.Where(t.DeclaringType.GetRuntimeMethods(), m => m.HasSameMetadataDefinitionAs(methodBase)))
```

Comment:
This use of `Enumerable.Where` through static-method style rather than the extension style is unidiomatic, but I decided to change this and a few similar cases after this is in rather than combining that change into this PR.